### DISCUSSION TITLE
Last status of New storage in Keycloak

### DISCUSSION THREAD

**Discussion by cw-20021423** [keycloak/keycloak#20715] [Archive]
Here is the last notice I could found about new store in keycloak. 
https://www.keycloak.org/2022/07/storage-map.html

Is there any updates to it? 

1. It is written that Keycloak 19 brings in an early preview of the future Keycloak store supporting Postgres and CockroachDB databases. How about latest version? does latest version support mysql ? If not is there any plan for supporting or is there any guide on data migration from other databases to supported one?

2. It says that since the new store is not yet production ready, it remains the default store in Keycloak 19. How about latest version? is new store not production ready yet? and which store is the default store on latest version?

3. What version of keycloak is more stable for production? There is no guide about it! 

4. It is mentioned that to configure Keycloak 19 with the new store can use --storage=chm in running command. how about keycloak 20 or 21? if the default store is new store then how to configure legacy store? 

5. The is no document of comparing legacy and new store. How did it changed? 

Thank you

**Comment by hmlnarik** [comment]
The default store is currently the legacy store.

The new store and is not yet production ready. Due to Keycloak priority shifts, its development has slowed in the recent months, and it will thus not be production ready in 22. Still, it is there since 19 to obtain the feedback from the community which is very welcome (#9596 #12010 #16768). Documentation of migration / function parity would be part of the docs.

**Reply by cw-20021423** [reply]
thank you for your reply~ you said not be production ready in 22, do you mean 2023?

**Reply by hmlnarik** [reply]
Keycloak versions got to high numbers... I mean Keycloak version 22
